modules/security/src/test/java/org/apache/ignite/internal/processor/security/AbstractRemoteSecurityContextCheckTest.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processor.security;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.BiFunction;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCompute;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.util.typedef.T2;
+import org.apache.ignite.internal.util.typedef.X;
+import org.apache.ignite.plugin.security.SecurityException;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ *
+ */
+public abstract class AbstractRemoteSecurityContextCheckTest extends AbstractSecurityTest {
+    /** Verifier to check results of tests. */
+    private static final Verifier VERIFIER = new Verifier();
+
+    /**
+     * Registers current security context and incriments invoke's counter.
+     */
+    protected static void register() {
+        VERIFIER.register((IgniteEx)Ignition.localIgnite());
+    }
+
+    /**
+     * @return IgniteCompute is produced by passed node for cluster group that contains nodes with ids from collection.
+     */
+    protected static IgniteCompute compute(Ignite ignite, Collection<UUID> ids) {
```

Comment:
Why `VERIFIER`, `compute`, `secSubjectId` are static?

Reply:
The cause of it is the guarantee that Verifier will be created only one time.

Reply:
We can't do these things not static because they use on remote nodes.
Don't require any changes.